src/atoms/sdp_cone/geom_mean_hypocone.jl

```diff
@@ -0,0 +1,169 @@
+#############################################################################
+# geom_mean_hypocone.jl
+# Constrains T to
+#   A #_t B ⪰ T
+# where:
+#   * A #_t B is the t-weighted geometric mean of A and B:
+#     A^{1/2} (A^{-1/2} B A^{-1/2})^t A^{1/2}
+#     Parameter t should be in [0,1].
+#   * Constraints A ⪰ 0, B ⪰ 0 are added.
+#
+# Note on parameter fullhyp:
+#   In many applications one doesn't need the full hypograph
+#     hyp_t = {(A,B,T) : A #_t B ⪰ T}
+#   but rather it is enough to work with a convex set C_t that satisfies
+#     (A,B,A #_t B) \in C_t
+#     (A,B,T) \in C_t  =>  A #_t B ⪰ T
+#   In this case one should set fullhyp = false. The SDP description will be
+#   (slightly) smaller. (By default fullhyp is set to true).
+#
+# All expressions and atoms are subtypes of AbstractExpr.
+# Please read expressions.jl first.
+#
+#REFERENCE
+#   Ported from CVXQUAD which is based on the paper: "Lieb's concavity
+#   theorem, matrix geometric means and semidefinite optimization" by Hamza
+#   Fawzi and James Saunderson (arXiv:1512.03401)
+#############################################################################
+
+struct GeomMeanHypoCone
+    A::AbstractExpr
+    B::AbstractExpr
+    t::Rational
+    size::Tuple{Int, Int}
+    fullhyp::Bool
+
+    function GeomMeanHypoCone(A::AbstractExpr, B::AbstractExpr, t::Rational, fullhyp::Bool=true)
+        if size(A) != size(B)
+            error("A and B must be the same size")
```

Comment:
Would be great to get some `@test_throws` tests to check these error cases! (the uncovered lines are marked by codecov)

Reply:
```suggestion
            throw(ArgumentError("`A` and `B` must be the same size"))
```